File: cpp/Sort/bubbleSort.cpp
#include "Sort.h"

int Sort::bubbleSort(vector<int> &nums, bool asc) {
	for (int i = 1; i < nums.size(); i++) {
		for (int j = 1; j < nums.size() + 1 - i; j++) {
			if (nums[j] < nums[j - 1]) swap(&nums[j], &nums[j - 1]);
		}
	}
	if (!asc) Sort::reverse(nums);
	return 0;
}

File: js/Sort/bubbleSort.js
function swap(arr, i, j) {
    var tmp = arr[i];
    arr[i] = arr[j];
    arr[j] = tmp;
    return arr
}

function reverse(nums) {
    for (var i=0; i<parseInt(nums.length / 2); i++) nums = swap(nums, i, nums.length-1-i);
    return nums;
}

function bubbleSort(nums, asc) {
    for (var i=1; i<nums.length; i++) {
        for (var j=1; j<nums.length-i+1; j++) {
            if (nums[j] < nums[j-1]) [nums[j], nums[j-1]] = [nums[j-1], nums[j]];
        }
    }
    if (asc !== true) nums = reverse(nums);
    return nums;
}

var res = bubbleSort([2,4,1,2], false)
console.log(res)
